src/network.cpp

```diff
@@ -38,6 +43,22 @@ String uptimeString() {
   return ret;
 }
 
+void retrieveAvailableWifiNetworks() {
+  // preset WIFI power setting to retrieve as many wifi networks as possible
+  WiFi.setOutputPower(17);
+  WiFi.mode(WIFI_STA);
+  WiFi.disconnect();
+  int n = WiFi.scanNetworks(false,false, 0, NULL);
```

Comment:
This is a blocking scan. I'm not even sure if it disables interrupts. If it does, this would interrupt the serial data processing. I tried pretty hard to make all of Owie asynchronous, I'll have to ask you to make this PR asynchronous as well.

I'd suggest rewriting this CL so that scans done automatically in the background as long as the Owie AP has a client connected to it. WiFi page could then just display the result of the last scan without blocking on anything.